commands/db.bee.inc

```diff
@@ -110,11 +119,71 @@ function db_import_bee_callback($arguments, $options) {
   }
 }
 
+/**
+ * Command callback: Drop and re-create the existing configured database.
+ */
+function db_drop_bee_callback($arguments, $options) {
+  global $_bee_yes_mode;
+  // Initialize our pipes variable.
+  $pipes = array();
+
+  // Get database info.
+  $db_connection = Database::getConnectionInfo();
+  $db_info = $db_connection['default'];
+
+  if (!$_bee_yes_mode) {
+    // Prompt to continue.
+    if (!bee_confirm(bt('Are you sure you want to drop the contents of !database ?', array(
+        '!database' => $db_info['database']
+      )), FALSE)) {
+      return;
+    }
+  }
+
+  $options = '--user=' . rawurldecode($db_info['username']);
+  $options .= " --password='" .  rawurldecode($db_info['password']) . "'";
+  $options .= ' --host=' . $db_info['host'];
+  $db_database = rawurldecode($db_info['database']);
+
+  // Drop the existing backdrop database as configured.
+  $command = 'mysql ' . $options . ' -e "drop database ' . $db_database . '";';
+  $result = proc_close(proc_open($command, array(STDIN, STDOUT, STDERR), $pipes));
+
+  if ($result == -1) {
+    bee_message(bt("The '!database' database could not be dropped.", array(
+      '!database' => $db_info['database'],
+    )), 'error');
+  }
+  else {
+    bee_message(bt("The '!database' database was successfully dropped.", array(
+      '!database' => $db_info['database'],
+    )), 'success');
+  }
+
+  // Re-create the existing backdrop database as configured.
+  $command = 'mysql ' . $options . ' -e "create database ' . $db_database . '";';
+  $result = proc_close(proc_open($command, array(STDIN, STDOUT, STDERR), $pipes));
+
+  if ($result == -1) {
+    bee_message(bt("The '!database' database could not be created.", array(
+      '!database' => $db_info['database'],
+    )), 'error');
+  }
+  else {
+    bee_message(bt("The '!database' database was successfully created.", array(
+      '!database' => $db_info['database'],
+    )), 'success');
+  }
+}
```

Comment:
Like we do in the other functions (from #261), we can delete the temporary mysql options file:

```suggestion
  }
  // Remove the temporary mysql options file.
  bee_delete($connection_file);  
}
```